--- origsrc/glom-1.12.3/macros/mm-python.m4	2009-11-02 03:23:25.000000000 -0600
+++ src/glom-1.12.3/macros/mm-python.m4	2009-11-03 03:32:30.408517700 -0600
@@ -67,6 +67,7 @@ dnl On Windows the library is in libs/, 
   for mm_dir
   do
     AS_IF([test -f "$mm_dir/lib$mm_pylib.so" || \
+           test -f "$mm_dir/lib$mm_pylib.dll.a" || \
            test -f "$mm_dir/lib$mm_pylib.a"],
           [AS_CASE([$mm_dir], [[/usr/lib|/usr/lib64]],
                    [PYTHON_LIBS="$PYTHON_LIBS -l$mm_pylib"; break],
