--- origsrc/glom-1.18.2/glom/main.cc	2011-06-17 09:27:26.000000000 -0500
+++ src/glom-1.18.2/glom/main.cc	2011-06-26 07:15:51.225246900 -0500
@@ -371,18 +371,18 @@ bool check_pygda_is_available_with_warni
 
 } //namespace Glom
 
-#ifndef G_OS_WIN32
+#ifndef G_PLATFORM_WIN32
 extern "C" void __libc_freeres(void);
 #endif
 
 int
 main(int argc, char* argv[])
 {
-#ifndef G_OS_WIN32
+#ifndef G_PLATFORM_WIN32
   //Force some cleanup at exit,
   //to help valgrind to detect memory leaks:
   atexit(__libc_freeres);
-#else
+#elif defined G_OS_WIN32
   WSADATA data;
   int errcode = WSAStartup(MAKEWORD(2, 0), &data);
   if(errcode != 0)
