--- origsrc/glom-1.20.2/glom/application.cc	2011-12-13 04:01:26.000000000 -0600
+++ src/glom-1.20.2/glom/application.cc	2011-12-21 21:47:12.356804300 -0600
@@ -1865,7 +1865,7 @@ bool Application::recreate_database_from
 
     //Tell the user:
     Gtk::Dialog* dialog = 0;
-    Utils::get_glade_widget_with_warning("glom_developer.glade", "dialog_error_create_database", dialog);
+    Utils::get_glade_widget_with_warning("developer/dialog_error_create_database.glade", "dialog_error_create_database", dialog);
     dialog->set_transient_for(*this);
     Glom::Utils::dialog_run_with_help(dialog, "dialog_error_create_database");
     delete dialog;
--- origsrc/glom-1.20.2/glom/frame_glom.cc	2011-12-13 04:00:30.000000000 -0600
+++ src/glom-1.20.2/glom/frame_glom.cc	2011-12-21 21:49:45.637015900 -0600
@@ -2264,7 +2264,7 @@ bool Frame_Glom::create_database(const G
   {
     //Tell the user:
     Gtk::Dialog* dialog = 0;
-    Utils::get_glade_widget_with_warning("glom_developer.glade", "dialog_error_create_database", dialog);
+    Utils::get_glade_widget_with_warning("developer/dialog_error_create_database.glade", "dialog_error_create_database", dialog);
     dialog->set_transient_for(*pWindowApp);
     Glom::Utils::dialog_run_with_help(dialog, "dialog_error_create_database");
     delete dialog;
